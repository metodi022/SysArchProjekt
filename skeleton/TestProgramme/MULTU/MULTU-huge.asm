lui $1, 0x8000  // 32'd2147483648
ori $1, 0x0000
lui $2, 0xbfff  // 32'd3221225471
ori $2, 0xffff
multu $1, $2    // multiply $1 and $2
mfhi $3         // 0x5fffffff
mflo $4         // 0x80000000

// (2^31)*((3*2^30)-1) = 2147483648 * 3221225471 = 6917529025493598208
//                     = 0x80000000 * 0xbfffffff = 0x5fffffff80000000
// expected: 0x5fffffff
