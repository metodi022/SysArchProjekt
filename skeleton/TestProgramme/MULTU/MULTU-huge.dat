0x3c018000
0x34210000
0x3c02bfff
0x3442ffff
0x00220019
0x00001810
0x00002012
