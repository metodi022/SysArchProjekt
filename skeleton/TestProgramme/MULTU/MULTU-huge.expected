@1 80000000
@2 bfffffff
@3 5fffffff
@4 80000000
